lib/bpf/map_create.h

```diff
@@ -0,0 +1,36 @@
+#ifndef LIB_BPF_MAP_CREATE_H_
+#define LIB_BPF_MAP_CREATE_H_
+
+#include <cstring>
+#include <string>
+#include <string_view>
+
+#include "lib/bpf/bpf.h"
+#include "lib/error/assign_or_return.h"
+#include "lib/error/status_or.h"
+#include "lib/posix/unique_file_descriptor.h"
+
+namespace bpf {
+
+// BPF_MAP_CREATE wrapper.
+// Create a map of 'map_type' from keys of type KeyT to values of type ValueT
+// and maximum capacity of 'max_entries'.
+//
+// Return an owned file descriptor on success or an error status on failure.
+//
+// See 'man 2 bpf' for details.
+template <typename KeyT, typename ValueT>
+inline error::StatusOr<posix::UniqueFileDescriptor> MapCreate(
+    const uint32_t map_type, const size_t max_entries) {
```

Comment:
Same question with respect to map_type. Do we want to have stronger typing at this level already?

Reply:
In terms of consistency, we are already exposing a more specific return type in this instance (when compared to the Bpf() call)

Reply:
See the comment on 'int cmd' and let me know what you think